### PR TITLE
tidb_query: Migrate json_type to RPN Function.

### PR DESCRIPTION
**Pull request by mapleFU**
## What have you changed? (mandatory)

* Create `impl_json.rs` in `components/tidb_query` 
* Migrate implementation for json_type from `builtin_json.rs` to the new `impl_json.rs`.
* Add support for `JsonTypeSig` in `rpn_expr/mod.rs` for `components/tidb_query`.

Please **NOTE** that:
- Do not assume reviewers understand the original issue

## What are the type of the changes? (mandatory)

- New feature (Support new RPN Function).

## How has this PR been tested? (mandatory)

New unit tests.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

File: components/tidb_query/src/rpn_expr/impl_json.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+use tidb_query_codegen::rpn_fn;
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_json.rs]: missing header

File: components/tidb_query/src/rpn_expr/impl_json.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#[inline]
+fn json_type(arg: &Option<Json>) -> Result<Option<Bytes>> {
+    Ok(arg
+        .as_ref()
+        .map(|json_arg| Bytes::from(json_arg.json_type())))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::super::super::expr::EvalContext;
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_json.rs]: use tidb_query::expr::EvalContext;

**Comment by breeswish** [comment]
CI failed

File: components/tidb_query/src/rpn_expr/impl_json.rs (hunk 2 above)
Comment (mapleFU) [reply components/tidb_query/src/rpn_expr/impl_json.rs]: > use crate::expr::EvalContext;

`EvalContext` has been removed now.

**Pull request merged**